frameworks/shared/serialization.py

```diff
@@ -28,16 +32,71 @@ def _import_data_libraries():
     pandas_compression=os.environ.get('AMLB_SER_PD_COMPR') or 'infer',
     # 'snappy', 'gzip', 'brotli', None ?
     pandas_parquet_compression=os.environ.get('AMLB_SER_PD_PQT_COMPR') or None,
+    # if numpy can use pickle to serialize ndarrays,
+    numpy_pickle=True,
+    # if sparse matrices should be compressed during serialization.
+    sparse_matrix_compression=True,
+    # if sparse matrices should be deserialized to some specific format:
+    #  None (no change), 'array' (numpy), 'dense' (dense matrix).
+    sparse_matrix_deserialized_format='array',
```

Comment:
Looks good to me, but if I understand correctly then deserialization in a sparse format is opt-in instead of opt-out (since the default is to deserialize to array)? Maybe we should make it opt-out instead, i.e. by default deserialize in sparse format unless specified otherwise in the `__init__.py`? Because I would argue that the default behavior should be that we can throw anything at the automl frameworks.

Reply:
I'm still working on this but Python frameworks accepting pandas DataFrame (instead of numpy arrays) also fail with sparse dataframes, so I'd rather make the default to "dense" or "array" for now ("dense" looks actually slightly better).
When we'll have full support for sparse matrices then  we may change the default, but for now it's compromised; for example there are problems with sparse dataframes and parquet format, pandas/arrows fails saving those to parquet format (https://github.com/apache/arrow/issues/1894).

Reply:
That looks like a good reason to have the default as you suggested. 👍

Reply:
changed my mind, set the default to `None` (sparse remains sparse) and explicitly set the sparse strategy per frameowkr if they need one.